src/a11y-dialog.ts

```diff
@@ -52,7 +52,7 @@ export default class A11yDialog {
 
     // Keep a reference to the currently focused element to be able to restore
     // it later
-    this.previouslyFocused = document.activeElement as HTMLElement
+    this.previouslyFocused = getDeepActiveElement() as HTMLElement
```

Comment:
🔍 We probably want `previouslyFocused` to be shadow-aware, to account for possible custom elements in useerland.